charmstore/fakeclient.go

```diff
@@ -0,0 +1,582 @@
+// Copyright 2019 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package charmstore
+
+import (
+	"bytes"
+	"crypto/sha512"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"time"
+
+	"github.com/juju/errors"
+	"github.com/juju/juju/api/charms"
+	"github.com/juju/juju/core/model"
+	"gopkg.in/juju/charm.v6"
+	charmresource "gopkg.in/juju/charm.v6/resource"
+	"gopkg.in/juju/charmrepo.v2"
+	"gopkg.in/juju/charmrepo.v3/csclient"
+	"gopkg.in/juju/charmrepo.v3/csclient/params"
+	"gopkg.in/macaroon.v1"
+)
+
+// fakeclient.go provides a charmstore client that does not use any HTTP connections
+// or authentication mechanisms
+
+type miniHTTP interface {
+	Get(path string, value interface{}) error
+	Put(path string, value interface{}) error
+}
+
+type CharmStorer interface {
+	miniHTTP
+	UploadCharm(id *charm.URL, ch charm.Charm) (*charm.URL, error)
```

Comment:
How many of these are actually needed for deploy?

Reply:
I've removed these "global" interfaces